app/src/main/java/com/monkeyteam/chimpagne/ui/event/details/supplies/StaffSupplyDialog.kt

```diff
@@ -0,0 +1,131 @@
+                      deleteSupply()
+                      onDismissRequest()
+                    }))
+  }
+
+  var displayEditSupplyDialog by remember { mutableStateOf(false) }
+  if (displayEditSupplyDialog) {
+    EditSupplyDialog(
+        supply = tempSupply,
+        onDismissRequest = { displayEditSupplyDialog = false },
+        onSave = { tempSupply = it })
+  }
+
+  CustomDialog(
+      title = "${tempSupply.quantity} ${tempSupply.unit}",
+      description = tempSupply.description,
+      onDismissRequest = onDismissRequest,
+      modifier = Modifier.testTag("staff_supply_dialog"),
+      buttonDataList =
+          listOf(
+              ButtonData(
+                  stringResource(id = R.string.chimpagne_cancel),
+                  onClick = onDismissRequest,
+                  modifier = Modifier.testTag("cancel_supply_button")),
+              ButtonData(
+                  stringResource(id = R.string.chimpagne_delete),
+                  modifier = Modifier.testTag("delete_supply_button")) {
+                    displayDeleteSupplyDialog = true
+                  },
+              ButtonData(
+                  stringResource(id = R.string.chimpagne_edit),
+                  modifier = Modifier.testTag("edit_supply_button")) {
+                    displayEditSupplyDialog = true
+                  },
+              ButtonData(
+                  stringResource(id = R.string.chimpagne_save),
+                  modifier = Modifier.testTag("save_supply_button")) {
+                    displayEditSupplyDialog = false
+                    updateSupply(tempSupply)
+                    onDismissRequest()
+                  })) {
+        LazyColumn(modifier = Modifier.fillMaxWidth()) {
+          if (tempSupply.assignedTo.keys.isEmpty()) {
+            item { Text(stringResource(id = R.string.supplies_supply_not_assigned)) }
+          } else {
+            item {
+              Text(
+                  stringResource(id = R.string.supplies_supply_assigned_to),
+                  modifier = Modifier.testTag("staff_list"))
+            }
+            accounts.entries.forEach { (userUID, userAccount) ->
+              if (tempSupply.assignedTo[userUID] == true) {
+                item {
+                  SupplyDialogAccountEntry(
+                      account = userAccount,
+                      loggedUserUID = loggedUserUID,
+                      showCheckBox = true,
+                      true) {
+                        tempSupply = tempSupply.copy(assignedTo = tempSupply.assignedTo - userUID)
+                      }
+                }
+              }
+            }
+          }
+          if (!tempSupply.assignedTo.keys.containsAll(accounts.keys)) {
+            item { Text(stringResource(id = R.string.supplies_not_assigned_to)) }
+            accounts.entries.forEach { (userUID, userAccount) ->
+              if (tempSupply.assignedTo[userUID] != true) {
```

Comment:
!= true should be changed to !tempSupply.assignedTo[userUID]

Reply:
tempSupply.assignedTo: Map<ChimpagneAccountUID, Boolean?> thus !tempSupply.assignedTo[userUID] doesn't work as tempSupply.assignedTo[userUID] can be null